README.md

```diff
@@ -1,5 +1,328 @@
 # Extism Rust PDK
 
-For more information about the Rust PDK, please [visit the docs](https://extism.org/docs/write-a-plugin/rust-pdk).
+[![crates.io](https://img.shields.io/crates/v/extism_pdk.svg)](https://crates.io/crates/extism-pdk)
+
+This library can be used to write [Extism Plug-ins](https://extism.org/docs/concepts/plug-in) in Rust.
+
+## Install
+
+Make sure you generate a `lib` project:
+
+```bash
+cargo new --lib my-plugin
+```
+
+Add the library from [crates.io](https://crates.io/crates/extism-pdk).
+
+```bash
+cargo add extism-pdk
+```
+
+Change your `Cargo.toml` to set the crate-type to `cdylib`:
+
+```toml
+[lib]
+crate_type = ["cdylib"]
+```
+
+## Getting Started
+
+The goal of writing an [Extism plug-in](https://extism.org/docs/concepts/plug-in) is to compile the your rust code to a Wasm module with some special export functions that the host application can invoke. The first thing you should understand is creating an export. Let's write a simple program that exports a `greet` function which will take a name as a string and return a greeting string:
+
+
+```rust
+use extism_pdk::*;
+
+#[plugin_fn]
+pub fn greet(name: String) -> FnResult<String> {
+    Ok(format!("Hello, {}!", name))
+}
+```
+
+Since we don't need any system access for this, we can compile this to the lightweight `wasm32-unknown-unknown` target instead of using the `wasm32-wasi` target:
+
```

Comment:
Is it worth mentioning that you can also set wasm32-unknown-unknown to be the default target by creating a `.cargo/config.toml`:

```
[build]
target = "wasm32-unknown-unknown"
```